controllers/workspace/devworkspace_controller.go

```diff
@@ -491,8 +491,10 @@ func (r *DevWorkspaceReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	reconcileStatus.setConditionTrue(conditions.DeploymentReady, "DevWorkspace deployment ready")
 
 	serverReady, serverStatusCode, err := checkServerStatus(clusterWorkspace)
-	if err != nil {
-		return reconcile.Result{}, err
+	if shouldReturn, reconcileResult, reconcileErr := r.checkDWError(workspace, err, "Error checking server status", metrics.ReasonInfrastructureFailure, reqLogger, &reconcileStatus); shouldReturn {
+		reqLogger.Info("Waiting for DevWorkspace to be ready")
+		reconcileStatus.setConditionFalse(dw.DevWorkspaceReady, "Waiting for DevWorkspace to be ready") // No sure of the conditions.DeploymentReady is the right one
```

Comment:
My mistake, I should have caught this earlier: please remove the comment  `// No sure of the conditions.DeploymentReady is the right one`. You can remove the comment, then `git add controllers/workspace/devworkspace_controller.go && git commit --amend && git push -f`